src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -556,13 +556,14 @@ private BoundExpression MakeImplicitConversion(BoundExpression rewrittenOperand,
 
                 TypeSymbol userDefinedConversionRewrittenType = conversion.Method.ReturnType;
 
-                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0])
+                // the conversion only needs to happen for non-nullable valuetypes
+                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0] && !userDefinedConversionRewrittenType.CanBeAssignedNull())
```

Comment:
It isn't clear to me that this type cannot be a type parameter. If it can, what is the right behavior for that case? This test doesn't seem apropos for that.